From 8f6868e3f27cb962bcb78857cab8f0458a9c9093 Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Thu, 2 Jul 2020 20:39:06 +0800
Subject: [PATCH] drm: Add DRM_CLIENT_CAP_SHARE_PLANES ioctl

this is only used to compatibility with old hwc code

Change-Id: Idcf2ca235d320656adf20d010dcd3a3940159c0d
Signed-off-by: Sandy Huang <hjc@rock-chips.com>
---
 drivers/gpu/drm/drm_ioctl.c | 2 ++
 include/uapi/drm/drm.h      | 7 +++++++
 2 files changed, 9 insertions(+)

diff --git a/drivers/gpu/drm/drm_ioctl.c b/drivers/gpu/drm/drm_ioctl.c
index d3fa7a0645af..925d9f4a1c91 100644
--- a/drivers/gpu/drm/drm_ioctl.c
+++ b/drivers/gpu/drm/drm_ioctl.c
@@ -318,6 +318,8 @@ drm_setclientcap(struct drm_device *dev, void *data, struct drm_file *file_priv)
 			return -EINVAL;
 		file_priv->universal_planes = req->value;
 		break;
+	case DRM_CLIENT_CAP_SHARE_PLANES:
+		break;
 	case DRM_CLIENT_CAP_ATOMIC:
 		if (!drm_core_check_feature(dev, DRIVER_ATOMIC))
 			return -EINVAL;
diff --git a/include/uapi/drm/drm.h b/include/uapi/drm/drm.h
index 300f336633f2..3acb728feaa7 100644
--- a/include/uapi/drm/drm.h
+++ b/include/uapi/drm/drm.h
@@ -696,6 +696,13 @@ struct drm_get_cap {
  */
 #define DRM_CLIENT_CAP_WRITEBACK_CONNECTORS	5
 
+/*
+ * DRM_CLIENT_CAP_SHARE_PLANES
+ *
+ * this is only used to compatibility with old hwc code.
+ */
+#define DRM_CLIENT_CAP_SHARE_PLANES	6
+
 /** DRM_IOCTL_SET_CLIENT_CAP ioctl argument type */
 struct drm_set_client_cap {
 	__u64 capability;
-- 
2.35.3

